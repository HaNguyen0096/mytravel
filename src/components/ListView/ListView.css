.logs{
  background-color: #A9A9A9;
  width: 70%;
  display: block;
  margin: 50px auto 50px auto ;
  text-align: center;
  border-style: double;
}

.logImage{
  height: 100px;
  width: auto;
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-top: 15px;
}

.Btn{
  margin-top: 10px;
  margin-bottom: 10px;
}

@media only screen and (min-width: 480px) {
  .logs{
    background-color: #A9A9A9;
    width: 70%;
    display: block;
    margin: 50px auto 50px auto ;
    text-align: center;
    border-style: double;
  }
  
  .logImage{
    height: 300px;
    width: auto;
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 15px;
  }
  
  .Btn{
    margin-top: 10px;
    margin-bottom: 10px;
  }
}