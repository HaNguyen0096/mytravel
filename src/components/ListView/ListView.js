import React, { Component } from 'react'
import ViewLogBtn from '../ViewLogBtn/ViewLogBtn'
import './ListView.css'

export default class ListView extends Component{
  
  render(){
    const {log} = this.props
    console.log(log)
    return (
      <div className='logs'>
        {log.image && <img className='logImage' src={log.image} alt={log.title}/>}
        <h1>{log.title}</h1>
        <div className='Btn'>
          <ViewLogBtn id={log.id}/>
        </div>
        
      </div>
    )
  }
}