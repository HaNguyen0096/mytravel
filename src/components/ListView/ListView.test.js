import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ListView from './ListView'

describe('ListView component', () => {

  const testLog = [
    {
      id: 1,
      latitude: '42.3554',
      longitude: '-71.0640',
      title: 'Boston Common',
      description: 'This is a beautiful place',
      image: 'https://en.wikipedia.org/wiki/File:Aerial_View_Parkman_Bandstand_at_Boston_Common.jpg',
      rating: 10,
      visited_day: '2020-03-18T07:55:12.825Z',
      public: false,
      date_created: '2020-03-18T07:55:12.825Z',
    }
  ]
  
  it('renders ListView without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <ListView log={testLog}/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for ListView as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <ListView log={testLog}/>
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders ListView by default', () => {
    const wrapper = shallow(<ListView  log={testLog}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})