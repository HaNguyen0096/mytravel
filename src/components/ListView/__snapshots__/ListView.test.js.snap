// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView component renders ListView by default 1`] = `
<div
  className="logs"
>
  <h1 />
  <div
    className="Btn"
  >
    <ViewLogBtn />
  </div>
</div>
`;

exports[`ListView component renders the UI for ListView as expected 1`] = `
<div
  className="logs"
>
  <h1 />
  <div
    className="Btn"
  >
    <div>
      <button
        className="ant-btn Btn ant-btn-primary"
        onClick={[Function]}
        type="button"
      >
        <a
          href="/logs/undefined"
          onClick={[Function]}
        >
          View More
        </a>
      </button>
    </div>
  </div>
</div>
`;
